extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/handlers/DevClasspathStaticHandler.java

```diff
@@ -0,0 +1,176 @@
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.impl.MimeMapping;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * {@link StaticHandler} implementation to handle static resources using the Classpath.
+ * This is meant to be used on {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT} mode.
+ */
+public class DevClasspathStaticHandler implements Handler<RoutingContext> {
+
+    private static final Logger LOG = Logger.getLogger(DevClasspathStaticHandler.class);
+    private static final int HTTP_STATUS_OK = 200;
+    private static final int HTTP_STATUS_NO_CONTENT = 204;
+    private static final String ALLOW_HEADER = "Allow";
+    private static final String ALLOW_HEADER_VALUE = "HEAD,GET,OPTIONS";
+    private final Set<String> generatedResources;
+    private final Set<String> compressedMediaTypes;
+    private final ClassLoader currentClassLoader;
+    private final boolean enableCompression;
+    private final String indexPage;
+    private final Charset defaultEncoding;
+
+    public DevClasspathStaticHandler(Set<String> generatedResources, DevClasspathStaticHandlerOptions options) {
+        this.generatedResources = generatedResources;
+        this.compressedMediaTypes = options.getCompressMediaTypes();
+        this.currentClassLoader = Thread.currentThread().getContextClassLoader();
+        this.enableCompression = options.isEnableCompression();
+        this.indexPage = options.getIndexPage();
+        this.defaultEncoding = options.getDefaultEncoding();
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+
+        String resolvedPath = resolvePath(context);
+        String path = resolvedPath.endsWith("/") ? resolvedPath.concat(this.indexPage) : resolvedPath;
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debugf("Handling request for path '%s'", path);
+        }
+
+        boolean containsGeneratedResource = this.generatedResources.contains(removeStartSlashes(path));
+
+        if (!containsGeneratedResource) {
+            beforeNextHandler(this.currentClassLoader, context);
+            return;
+        }
+
+        if (context.request().method().equals(HttpMethod.OPTIONS)) {
+            context.response().putHeader(ALLOW_HEADER, ALLOW_HEADER_VALUE).setStatusCode(HTTP_STATUS_NO_CONTENT)
+                    .send();
+            return;
+        }
+
+        compressIfNeeded(context, path);
+
+        Vertx vertx = context.vertx();
+        vertx.executeBlocking(future -> {
+            try {
+                byte[] content = getClasspathResourceContent(path);
+                future.complete(content);
+            } catch (Exception e) {
+                future.fail(e);
+            }
+        }, asyncResult -> {
+            if (asyncResult.succeeded()) {
```

Comment:
@mcruzdev here, this if should be replaced by `if(asyncResult.result() != null)`  you should add another if for the succeed and in case of failure:
```java
context.fail(res.cause());
```